utils/rpms/daos.spec

```diff
@@ -204,13 +214,40 @@ PREFIX="%{?_prefix}"
 mkdir -p %{?buildroot}/%{_sysconfdir}/ld.so.conf.d/
 echo "%{_libdir}/daos_srv" > %{?buildroot}/%{_sysconfdir}/ld.so.conf.d/daos.conf
 mkdir -p %{?buildroot}/%{_unitdir}
-install -m 644 utils/systemd/daos_server.service %{?buildroot}/%{_unitdir}
-install -m 644 utils/systemd/daos_agent.service %{?buildroot}/%{_unitdir}
+install -m 644 utils/systemd/%{server_svc_name} %{?buildroot}/%{_unitdir}
+install -m 644 utils/systemd/%{agent_svc_name} %{?buildroot}/%{_unitdir}
 
 %pre server
 getent group daos_admins >/dev/null || groupadd -r daos_admins
-%post server -p /sbin/ldconfig
-%postun server -p /sbin/ldconfig
+%post server
+/sbin/ldconfig
+if [ $1 -eq 1 ]; then
+  systemctl preset %{server_svc_name} >/dev/null 2>&1
+fi
```

Comment:
`%systemd_post` is defined to do this.

Reply:
In EL7, yes. I couldn't find any evidence that these macros are defined for opensuse though. So rather than conditionally defining them I figured I'd just use the same for both.

Also, when I was looking at those macros for EL7, it didn't seem like they were compatible with sub-package scriptlets.

Reply:
oh wait... seems like opensuse puts them in a different RPM. Let's see if that works.